src/filter.js

```diff
@@ -0,0 +1,36 @@
+const filterOption = ['-f --filter [option]', 'Filter packages where a command should run'];
+const applyFilter = (options, packages) => {
+	let filter;
+	let filteredPackages;
+	let parsedOptions = options;
+	if (Array.isArray(options)) {
+		const filterIndex = options.indexOf('--filter') || options.indexOf('-f');
+		if (filterIndex !== -1) {
+			filter = options[filterIndex + 1];
+			parsedOptions = [...options.slice(0, filterIndex), ...options.slice(filterIndex + 2)];
+		}
+	} else {
+		if (options.hasOwnProperty('filter')) {
+			filter = options.filter;
+		}
+	}
```

Comment:
I think Commander has all the tools to handle this for you, see my comment above about `command.opts()`